archunit/src/main/java/com/tngtech/archunit/lang/conditions/ArchConditions.java

```diff
@@ -1284,30 +1276,62 @@ public static ArchCondition<JavaCodeUnit> onlyBeCalledByConstructorsThat(Describ
+     *
+     * @param predicate The predicate determining which objects satisfy/violate the condition
+     * @return A {@link ConditionByPredicate ConditionByPredicate} derived from the predicate and with 'have'-descriptions
+     * @param <T> The type of object the condition will test, e.g. a {@link JavaClass}
+     */
+    @PublicAPI(usage = ACCESS)
+    public static <T extends HasDescription & HasSourceCodeLocation> ConditionByPredicate<T> have(DescribedPredicate<? super T> predicate) {
+        return ArchCondition.from(predicate.<T>forSubtype())
+                .as(ArchPredicates.have(predicate).getDescription())
+                .describeEventsBy((predicateDescription, satisfied) -> (satisfied ? "has " : "does not have ") + predicateDescription);
     }
 
-    private static final IsConditionByPredicate<JavaClass> BE_TOP_LEVEL_CLASSES =
-            new IsConditionByPredicate<>("a top level class", JavaClass.Predicates.TOP_LEVEL_CLASSES);
-    private static final IsConditionByPredicate<JavaClass> BE_NESTED_CLASSES =
-            new IsConditionByPredicate<>("a nested class", JavaClass.Predicates.NESTED_CLASSES);
-    private static final IsConditionByPredicate<JavaClass> BE_MEMBER_CLASSES =
-            new IsConditionByPredicate<>("a member class", JavaClass.Predicates.MEMBER_CLASSES);
-    private static final IsConditionByPredicate<JavaClass> BE_INNER_CLASSES =
-            new IsConditionByPredicate<>("an inner class", JavaClass.Predicates.INNER_CLASSES);
-    private static final IsConditionByPredicate<JavaClass> BE_ANONYMOUS_CLASSES =
-            new IsConditionByPredicate<>("an anonymous class", JavaClass.Predicates.ANONYMOUS_CLASSES);
-    private static final IsConditionByPredicate<JavaClass> BE_LOCAL_CLASSES =
-            new IsConditionByPredicate<>("a local class", JavaClass.Predicates.LOCAL_CLASSES);
+    /**
+     * Derives an {@link ArchCondition} from a {@link DescribedPredicate}. Similar to {@link ArchCondition#from(DescribedPredicate)},
+     * but more conveniently creates a message to be used within a 'be'-sentence.
+     * <br>
+     * Take e.g. {@code be(assignableTo(Demo.class))}, then the condition description would be {@code "be assignable to com.example.Demo"}
+     * and each violating event would be described as {@code "Class <some.Example> is not assignable to com.example.Demo in (Example.java:0)"}.
+     *
+     * @param predicate The predicate determining which objects satisfy/violate the condition
+     * @return A {@link ConditionByPredicate ConditionByPredicate} derived from the predicate and with 'be'-descriptions
+     * @param <T> The type of object the condition will test, e.g. a {@link JavaClass}
+     */
+    @PublicAPI(usage = ACCESS)
+    public static <T extends HasDescription & HasSourceCodeLocation> ConditionByPredicate<T> be(DescribedPredicate<? super T> predicate) {
```

Comment:
I see you are declaring the predicate contravariantly when I opted to rely on the ArchCondition being treated contravariantly already.
I have not compared any compile time issues for readability of the type errors - have you?

Reply:
Yes, I was getting your point, but when I did it non-contravariantly I got compile errors in other places then :disappointed: Or in other words, each way makes it less nice in some situation but nicer in others. Then I thought I might as well go with the consistent choice that doesn't require any exceptions in the common patterns :man_shrugging: That was the background for this...